File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/robotSpecific/targetLanguage/calliope2017NoBlue/math.cpp
#define _GNU_SOURCE

#include "MicroBit.h"
#include "NEPODefs.h"
#include <list>
#include <array>
#include <stdlib.h>
MicroBit _uBit;

void ____math();


inline bool _isPrime(double d);

double ___numberVar;
std::list<double> ___numberList;

int main()
{
    _uBit.init();
    ___numberVar = 0;
    ___numberList = {0, 0};

    ____math();
    release_fiber();
}

void ____math() {
    _uBit.display.scroll(ManagedString(0));
    _uBit.display.scroll(ManagedString(___numberVar + ___numberVar));
    _uBit.display.scroll(ManagedString(___numberVar - ___numberVar));
    _uBit.display.scroll(ManagedString(___numberVar * ___numberVar));
    _uBit.display.scroll(ManagedString(___numberVar / ((float) ___numberVar)));
    _uBit.display.scroll(ManagedString(pow(___numberVar, ___numberVar)));
    _uBit.display.scroll(ManagedString(sqrt(___numberVar)));
    _uBit.display.scroll(ManagedString(abs(___numberVar)));
    _uBit.display.scroll(ManagedString(- (___numberVar)));
    _uBit.display.scroll(ManagedString(log(___numberVar)));
    _uBit.display.scroll(ManagedString(log10(___numberVar)));
    _uBit.display.scroll(ManagedString(exp(___numberVar)));
    _uBit.display.scroll(ManagedString(pow(10.0, ___numberVar)));
    _uBit.display.scroll(ManagedString(sin(M_PI / 180.0 * (___numberVar))));
    _uBit.display.scroll(ManagedString(cos(M_PI / 180.0 * (___numberVar))));
    _uBit.display.scroll(ManagedString(tan(M_PI / 180.0 * (___numberVar))));
    _uBit.display.scroll(ManagedString(180.0 / M_PI * asin(___numberVar)));
    _uBit.display.scroll(ManagedString(180.0 / M_PI * acos(___numberVar)));
    _uBit.display.scroll(ManagedString(180.0 / M_PI * atan(___numberVar)));
    _uBit.display.scroll(ManagedString(M_PI));
    _uBit.display.scroll(ManagedString(M_E));
    _uBit.display.scroll(ManagedString(M_GOLDEN_RATIO));
    _uBit.display.scroll(ManagedString(M_SQRT2));
    _uBit.display.scroll(ManagedString(M_SQRT1_2));
    _uBit.display.scroll(ManagedString((fmod(___numberVar, 2) == 0)));
    _uBit.display.scroll(ManagedString((fmod(___numberVar, 2) != 0)));
    _uBit.display.scroll(ManagedString((___numberVar > 0)));
    _uBit.display.scroll(ManagedString((___numberVar < 0)));
    _uBit.display.scroll(ManagedString((fmod(___numberVar,___numberVar) == 0)));
    _uBit.display.scroll(ManagedString(round(___numberVar)));
    _uBit.display.scroll(ManagedString(ceil(___numberVar)));
    _uBit.display.scroll(ManagedString(floor(___numberVar)));
    _uBit.display.scroll(ManagedString((int) ___numberVar % ((int) ___numberVar)));
    _uBit.display.scroll(ManagedString(min(max(___numberVar, ___numberVar), ___numberVar)));
    _uBit.display.scroll(ManagedString((_uBit.random(___numberVar - ___numberVar + 1) + ___numberVar)));
    _uBit.display.scroll(ManagedString(((double) rand() / (RAND_MAX))));
    _uBit.display.scroll(ManagedString(_isPrime(___numberVar)));
    _uBit.display.scroll(ManagedString((___numberVar == floor(___numberVar))));
    ___numberVar += ___numberVar;
    _uBit.display.scroll(ManagedString(_getListSum(___numberList)));
    _uBit.display.scroll(ManagedString(_getListMin(___numberList)));
    _uBit.display.scroll(ManagedString(_getListMax(___numberList)));
    _uBit.display.scroll(ManagedString(_getListAverage(___numberList)));
    _uBit.display.scroll(ManagedString(_getListMedian(___numberList)));
    _uBit.display.scroll(ManagedString(_getListStandardDeviation(___numberList)));

}

inline bool _isPrime(double d) {
    if (!(d == floor(d))) {
        return false;
    }
    int n = (int)d;
    if (n < 2) {
        return false;
    }
    if (n == 2) {
        return true;
    }
    if (n % 2 == 0) {
        return false;
    }
    for (int i = 3, s = (int)(sqrt(d) + 1); i <= s; i += 2) {
        if (n % i == 0) {
            return false;
        }
    }
    return true;
}

File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/robotSpecific/targetLanguage/calliopev3/math.py
import calliopemini
import random
import math

class BreakOutOfALoop(Exception): pass
class ContinueLoop(Exception): pass

timer1 = calliopemini.running_time()


___numberVar = 0
___numberList = [0, 0]
def ____math():
    global timer1, ___numberVar, ___numberList
    calliopemini.display.scroll(str(0))
    calliopemini.display.scroll(str(___numberVar + ___numberVar))
    calliopemini.display.scroll(str(___numberVar - ___numberVar))
    calliopemini.display.scroll(str(___numberVar * ___numberVar))
    calliopemini.display.scroll(str(___numberVar / float(___numberVar)))
    calliopemini.display.scroll(str(math.pow(___numberVar, ___numberVar)))
    calliopemini.display.scroll(str(math.sqrt(___numberVar)))
    calliopemini.display.scroll(str(math.fabs(___numberVar)))
    calliopemini.display.scroll(str(- (___numberVar)))
    calliopemini.display.scroll(str(math.log(___numberVar)))
    calliopemini.display.scroll(str(math.log10(___numberVar)))
    calliopemini.display.scroll(str(math.exp(___numberVar)))
    calliopemini.display.scroll(str(math.pow(10, ___numberVar)))
    calliopemini.display.scroll(str(math.sin(___numberVar)))
    calliopemini.display.scroll(str(math.cos(___numberVar)))
    calliopemini.display.scroll(str(math.tan(___numberVar)))
    calliopemini.display.scroll(str(math.asin(___numberVar)))
    calliopemini.display.scroll(str(math.acos(___numberVar)))
    calliopemini.display.scroll(str(math.atan(___numberVar)))
    calliopemini.display.scroll(str(math.pi))
    calliopemini.display.scroll(str(math.e))
    calliopemini.display.scroll(str((1 + 5 ** 0.5) / 2))
    calliopemini.display.scroll(str(math.sqrt(2)))
    calliopemini.display.scroll(str(math.sqrt(0.5)))
    calliopemini.display.scroll(str((___numberVar % 2) == 0))
    calliopemini.display.scroll(str((___numberVar % 2) == 1))
    calliopemini.display.scroll(str(___numberVar > 0))
    calliopemini.display.scroll(str(___numberVar < 0))
    calliopemini.display.scroll(str((___numberVar % ___numberVar) == 0))
    calliopemini.display.scroll(str(round(___numberVar)))
    calliopemini.display.scroll(str(math.ceil(___numberVar)))
    calliopemini.display.scroll(str(math.floor(___numberVar)))
    calliopemini.display.scroll(str(( ( ___numberVar ) % ( ___numberVar ) )))
    calliopemini.display.scroll(str(min(max(___numberVar, ___numberVar), ___numberVar)))
    calliopemini.display.scroll(str(random.randint(___numberVar, ___numberVar)))
    calliopemini.display.scroll(str(random.random()))
    calliopemini.display.scroll(str(_isPrime(___numberVar)))
    calliopemini.display.scroll(str((___numberVar % 1) == 0))
    ___numberVar += ___numberVar
    calliopemini.display.scroll(str(sum(___numberList)))
    calliopemini.display.scroll(str(min(___numberList)))
    calliopemini.display.scroll(str(max(___numberList)))
    calliopemini.display.scroll(str(float(sum(___numberList))/len(___numberList)))
    calliopemini.display.scroll(str(_median(___numberList)))
    calliopemini.display.scroll(str(_standard_deviation(___numberList)))


def run():
    global timer1, ___numberVar, ___numberList
    ____math()

def main():
    try:
        run()
    except Exception as e:
        raise

def _median(l):
    l = sorted(l)
    l_len = len(l)
    if l_len < 1:
        return None
    if l_len % 2 == 0:
        return (l[int((l_len - 1) / 2)] + l[int((l_len + 1) / 2)] ) / 2.0
    else:
        return l[int((l_len - 1) / 2)]

def _isPrime(number):
    if(number == 0 or number == 1):
        return False
    for i in range(2, int(math.floor(math.sqrt(number))) + 1):
        remainder = number % i
        if remainder == 0:
            return False
    return True

def _standard_deviation(l):
    mean = float(sum(l)) / len(l)
    sd = 0
    for i in l:
        sd += (i - mean)*(i - mean)
    return math.sqrt(sd / len(l))
if __name__ == "__main__":
    main()